.github/workflows/clang-tidy.yaml

```diff
@@ -0,0 +1,75 @@
+name: clang-tidy
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  clang-tidy:
+    runs-on: ubuntu-latest
```

Comment:
Better use:

runs-on: ubuntu-24.04

clang18 is already there.

Note: ubuntu-latest is only 22.04, because 24.04 is not marked as stable yet in github.

Reply:
Done.